pymongo/asynchronous/monitor.py

```diff
@@ -247,7 +247,9 @@ async def _check_server(self) -> ServerDescription:
             _sanitize(error)
             sd = self._server_description
             address = sd.address
-            duration = time.monotonic() - start
+            end = time.monotonic()
+            # time.monotonic() is not monotonic
+            duration = max(0.0, end - start)
```

Comment:
I like it but would you mind refactoring the 3 calls in this file like this:
```python
def _monotonic_duration(start: float) -> float:
    """Return the duration since the given start time.
    
    Accounts for buggy platforms where time.monotonic() is actually not monotonic. See PYTHON-4600.
    """
    return max(0.0, time.monotonic() - start))
```

Reply:
done